google/cloud/storage/internal/curl_client.cc

```diff
@@ -510,9 +510,10 @@ StatusOr<ObjectMetadata> CurlClient::InsertObjectMedia(
   }
 
   // If the application has set an explicit hash value we need to use multipart
-  // uploads.
-  if (!request.HasOption<DisableMD5Hash>() &&
-      !request.HasOption<DisableCrc32cChecksum>()) {
+  // uploads. `DisableMD5Hash` and `DisableCrc32cChecksum` should not be
+  // dependent on each other.
+  if (!request.GetOption<DisableMD5Hash>().value() ||
+      !request.GetOption<DisableCrc32cChecksum>().value_or(false)) {
     return InsertObjectMediaMultipart(request);
```

Comment:
I use `||` instead of `&&` because if one of there 2 options (`MD5` or `Crc32`) is disabled, the other one should still be computed as usual to prevent data corruptions.